mathesar/tests/api/test_constraint_api.py

```diff
@@ -423,3 +423,35 @@ def test_drop_nonexistent_table(client):
     response_data = response.json()[0]
     assert response_data['message'] == "Not found."
     assert response_data['code'] == ErrorCodes.NotFound.value
+
+
+def test_empty_column_list(create_patents_table, client):
+    table_name = 'NASA Constraint List 11'
+    table = create_patents_table(table_name)
+    data = {
+        'type': 'unique',
+        'columns': []
+    }
+    response = client.post(
+        f'/api/db/v0/tables/{table.id}/constraints/', data
+    )
+    response_data = response.json()[0]
+    assert response.status_code == 400
+    assert response_data['code'] == ErrorCodes.UnknownError.value
+    assert response_data['message'] == 'Columns field cannot be empty'
+
+
+def test_invalid_constrint_type(create_patents_table, client):
+    table_name = 'NASA Constraint List 12'
+    table = create_patents_table(table_name)
+    data = {
+        'type': 'foo',
+        'columns': [1]
+    }
+    response = client.post(
+        f'/api/db/v0/tables/{table.id}/constraints/', data
+    )
+    response_data = response.json()[0]
+    assert response.status_code == 400
+    assert response_data['code'] == ErrorCodes.UnknownDBType.value
+    assert 'Unknown database type identifier' in response_data['message']
```

Comment:
This test case is failing and the response_data for the sent request is `{'code': None, 'message': '', 'details': {'exception': "{'code': 'Unknown database type identifier foo.', 'message': 'None', 'field': None, 'details': None}"}}`